octobot_trading/exchanges/exchange_builder.py

```diff
@@ -118,6 +121,17 @@ async def _build_trading_modes(self, trading_mode_class):
             self.logger.error(f"An error occurred when initializing trading mode : {e}")
             raise e
 
+    def _ensure_trading_mode_compatibility(self, trading_mode_class):
+        to_check_exchange_type = enums.ExchangeTypes.SPOT
+        if self.exchange_manager.is_future:
+            to_check_exchange_type = enums.ExchangeTypes.FUTURE
+        if self.exchange_manager.is_margin:
+            to_check_exchange_type = enums.ExchangeTypes.MARGIN
```

Comment:
There is already a method doing approximately the same thing named `get_signal_exchange_type`, could we factorize them?

Reply:
right